Mage.Sets/src/mage/cards/l/LiarsPendulum.java

```diff
@@ -0,0 +1,142 @@
+
+    public LiarsPendulum(final LiarsPendulum card) {
+        super(card);
+    }
+
+    @Override
+    public LiarsPendulum copy() {
+        return new LiarsPendulum(this);
+    }
+}
+
+
+class LiarsPendulumEffect extends OneShotEffect {
+
+    public LiarsPendulumEffect() {
+        super(Outcome.DrawCard);
+        this.staticText = "Choose a card name. Target opponent guesses whether a card with that name is in your hand. You may reveal your hand. If you do and your opponent guessed wrong, draw a card";
+    }
+
+    public LiarsPendulumEffect(final LiarsPendulumEffect effect) {
+        super(effect);
+    }
+
+    @Override
+    public LiarsPendulumEffect copy() {
+        return new LiarsPendulumEffect(this);
+    }
+
+    @Override
+    public boolean apply(Game game, Ability source) {
+        Player controller = game.getPlayer(source.getControllerId());
+        Player opponent = game.getPlayer(this.getTargetPointer().getFirst(game, source));
+        if (controller != null && opponent != null) {
+            // Name a card.
+            Choice choice = new ChoiceImpl();
+            choice.setChoices(CardRepository.instance.getNames());
+            while (!controller.choose(Outcome.Benefit, choice, game)) {
+                if (!controller.canRespond()) {
+                    return false;
+                }
+            }
+            String cardName = choice.getChoice();
+            game.informPlayers("Card named: " + cardName);
+            boolean opponentGuess = false;
+            
+            if (opponent.chooseUse(Outcome.Neutral, "Is the chosen card (" + cardName + ") in " + controller.getLogName() + "'s hand?", source, game)) {
+                opponentGuess = true;
+            }
+            boolean rightGuess = !opponentGuess;
+            
+            for (Card card : controller.getHand().getCards(game)) {
+                if (card.isSplitCard()){
+                    SplitCard splitCard = (SplitCard) card;
+                    if (splitCard.getLeftHalfCard().getName().equals(cardName)){
+                        rightGuess = opponentGuess;
+                    }
+                    else if (splitCard.getRightHalfCard().getName().equals(cardName)){
+                        rightGuess = opponentGuess;
+                    }
+                }
+                if (card.getName().equals(cardName)) {
+                    rightGuess = opponentGuess;
+                }
+            }
+            game.informPlayers(opponent.getLogName() + " guesses that " + cardName + " is " + (opponentGuess ? "" : "not") + " in " + controller.getLogName() + "'s hand");
+            
+            if (controller.chooseUse(outcome, "Reveal your hand?", source, game)) {
+                controller.revealCards("hand of " + controller.getName(), controller.getHand(), game);
+                if (!rightGuess) {
+                    controller.drawCards(1, game);
+                    return true;
+                }
+            }
+        }
```

Comment:
the apply method should only return false if a not expected situation happens.
So if the player guesses the card name wrong, it should return true and not false, because this is a normal situation.
At least nothing special happens depending the return code.
Only if false is returned, there is a info message to the log to be able to recognize logical problems of an effects apply method.

Reply:
I've now moved the return to follow after the player choice.